cg/store/api/find_business_data.py

```diff
@@ -485,21 +487,34 @@ def get_samples_by_customer_id_and_pattern(
             filter_functions=filter_functions,
         ).all()
 
+    def get_samples_by_customer_and_subject_id_query(
+        self, customer_id: str, subject_id: str
+    ) -> Query:
+        """Return query of samples of customer with given subject_id or subject_id."""
+        records: Query = apply_customer_filter(
+            customers=self._get_join_sample_and_customer_query(),
+            customer_id=customer_id,
+            filter_functions=[CustomerFilter.FILTER_BY_INTERNAL_ID],
+        )
+        return apply_sample_filter(
+            samples=records,
+            subject_id=subject_id,
+            filter_functions=[SampleFilter.FILTER_BY_SUBJECT_ID],
+        )
```

Comment:
I think this is much clearer as shows two independent filtering steps, but is my personal opinion
```suggestion
        records: Query = apply_sample_filter(
            samples=records,
            subject_id=subject_id,
            filter_functions=[SampleFilter.FILTER_BY_SUBJECT_ID],
        )
        return records
```